lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -2800,6 +2800,105 @@ class DecomposeAtenRepeatOp : public OpRewritePattern<AtenRepeatOp> {
 };
 } // namespace
 
+// decompose aten.repeat_interleave.self_int into following ops:
+// aten.flatten.using_ints, aten.unsqueeze, aten.tile, aten.reshape
+namespace {
+
+class DecomposeAtenRepeatInterleaveSelfIntOp
```

Comment:
I think we can simplify this a bit by doing something like

```python
def my_repeat_interleave(input, repeats, dim=None):
    dim_is_none = dim is None
    if dim is None:
        dim = input.dim() - 1

    repeat_shape = [-1] * (input.dim() + 1)
    repeat_shape[dim + 1] = repeats
    input = input.unsqueeze(dim + 1)
    tiled = input.expand(repeat_shape)

    if dim_is_none:
        return tiled.flatten()
    else:
        return torch.ops.prims.collapse(tiled, dim, dim + 1)
```

If we use this approach with the `unsqueeze` helper mentioned by @zjgarvey, the only op that needs shape calculation is the `expand` op, and the shape for it is already computed as part of the decomposition (`repeat_shape`).

We are currently in a bit of an awkward transition where different users depend on different parts of the old and new stack in torch-mlir. In general I agree that we should only include shape calculations in decompositions if it does not get too in the way of readability or if it is deeply needed by the community. In my opinion, this case with the simplifications above is simple enough to not get in the way of readability.

Reply:
Thank you! I've implemented the changes you suggested. Could you please take another look at the code?